README.md

```diff
@@ -1,8 +1,361 @@
+
+The main goal of CCCL is to fill a similar role that the Standard C++ Library fills for Standard C++ – provide general-purpose, speed-of-light tools to CUDA C++ developers with general-purpose, speed-of-light tools to focus on solving the problems that matter.
+Unifying these projects is the first step towards realizing that goal. 
+
+## Example
+
+This is a simple example demonstrating the use of CCCL functionality from Thrust, CUB, and libcudacxx.
+
+It shows how to use Thrust/CUB/libcudacxx to implement a simple parallel reduction kernel. 
+Each thread block computes the sum of a subset of the array using `cub::BlockReduce`. 
+The sum of each block is then reduced to a single value using an atomic add via `cuda::atomic_ref` from libcudacxx.
+
+It then shows how the same reduction can be done using Thrust's `reduce` algorithm and compares the results.
+
+```cpp
+#include <thrust/device_vector.h>
+#include <cub/block/block_reduce.cuh>
+#include <cuda/atomic>
+#include <cstdio>
+
+constexpr int block_size = 256;
+
+__global__ void reduce(int const* data, int* result, std::size_t N) {
+  using BlockReduce = cub::BlockReduce<int, block_size>;
+  __shared__ typename BlockReduce::TempStorage temp_storage;
+
+  int index = threadIdx.x + blockIdx.x * blockDim.x;
+  int sum = 0;
+  if (index < N) {
+    sum += data[index];
+  }
+  sum = BlockReduce(temp_storage).Sum(sum);
+
+  if (threadIdx.x == 0) {
+    cuda::atomic_ref<int, cuda::thread_scope_device> atomic_result(*result);
+    atomic_result.fetch_add(sum, cuda::memory_order_relaxed);
+  }
+}
+
+int main() {
+
+  // Allocate and initialize input data
+  std::size_t N = 1000;
+  thrust::device_vector<int> data(N);
+  thrust::fill(data.begin(), data.end(), 1);
+
+  // Allocate output data
+  thrust::device_vector<int> kernel_result(1);
+
+  // Compute the sum reduction of `data` using a custom kernel
+  int num_blocks = (N + block_size - 1) / block_size;
+  reduce<<<num_blocks, block_size>>>(thrust::raw_pointer_cast(data.data()),
+                                     thrust::raw_pointer_cast(kernel_result.data()),
+                                     N);
+
+  auto err = cudaDeviceSynchronize();
+  if (err != cudaSuccess) {
+    std::cout << "Error: " << cudaGetErrorString(err) << std::endl;
+    return -1;
+  }
+
+  // Compute the same sum reduction using Thrust
+  int thrust_result = thrust::reduce(thrust::device, data.begin(), data.end(), 0);
+
+  // Ensure the two solutions are identical
+  std::printf("Custom kernel sum: %d\n", kernel_result[0]);
+  std::printf("Thrust reduce sum: %d\n", thrust_result);
+  assert(kernel_result[0] == thrust_result);
+  return 0;
+}
+```
+
+## Getting Started
+
+### Users
+
+Everything in CCCL is header-only. 
+Therefore, users need only concern themselves with how they get the header files and how they incorporate them into their build system. 
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick (non-blocking):** ‏ There are a lot of places like this where there is a newline where it isn't needed. The rendered result will be the same no matter what, but it would be nice to flow the text and use a consistent right margin for readers of the plain text version.